PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -12,6 +12,8 @@ class PurchasesTests: XCTestCase {
 
     override func setUp() {
         self.userDefaults = UserDefaults(suiteName: "TestDefaults")
+        requestFetcher = MockRequestFetcher()
+        systemInfo = MockSystemInfo(platformFlavor: nil, platformFlavorVersion: nil, finishTransactions: true)
```

Comment:
I think we should be doing this for all mocks - otherwise they just get created once and keep state in between tests. 
I'm just doing it here for these two since I had issues, but I'm not 100% it's necessary. I'll check before merging.